src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.hpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+
+/// \ingroup SurfacesGroup
+/// Contains functions that are used to find contour levels.
+namespace SurfaceFinder {
+/*!
+ * \ingroup SurfacesGroup
+
+ * \brief Function that interpolates data onto radial rays in the direction
+ * of the logical coordinates \f$\xi\f$ and \f$\eta\f$ and tries to perform a
+ * root find of \f$\text{data}-\text{target}\f$. Returns the logical
+ * coordinates of the roots
+ * found.
+ *
+ * \details We assume the element is part of a wedge block, so that the
+ * \f$\zeta\f$ logical coordinate points in the radial direction. Will fail if
```

Comment:
If I've understood the algorithm correctly, this restriction would be relatively straightforward to relax (replacing the hard-coded `0,1,2` with variables and passing in the direction). Could you add a sentence like `This can be generalized if necessary by passing in which logical direction to search for the surface.`?